packages/teleterm/src/ui/DocumentCluster/ClusterResources/Servers/Servers.tsx

```diff
@@ -36,43 +41,77 @@ export default function Container() {
 }
 
 function ServerList(props: State) {
-  const { servers = [], getSshLogins, connect, syncStatus } = props;
+  const {
+    servers = [],
+    getSshLogins,
+    connect,
+    fetchAttempt,
+    agentFilter,
+    pageCount,
+    customSort,
+    prevPage,
+    nextPage,
+    updateQuery,
+    onAgentLabelClick,
+    disabledRows,
+    updateSearch,
+  } = props;
   return (
     <>
-      {syncStatus.status === 'failed' && (
-        <Danger>{syncStatus.statusText}</Danger>
+      {fetchAttempt.status === 'error' && (
+        <Danger>{fetchAttempt.statusText}</Danger>
       )}
-      <Table
-        columns={[
-          {
-            key: 'hostname',
-            headerText: 'Hostname',
-            isSortable: true,
-          },
-          {
-            key: 'addr',
-            headerText: 'Address',
-            isSortable: true,
-            render: renderAddressCell,
-          },
-          {
-            key: 'labelsList',
-            headerText: 'Labels',
-            render: renderLabelCell,
-          },
-          {
-            altKey: 'connect-btn',
-            render: server =>
-              renderConnectCell(
-                () => getSshLogins(server.uri),
-                login => connect(server.uri, login)
-              ),
-          },
-        ]}
-        emptyText="No Nodes Found"
-        data={servers}
-        pagination={{ pageSize: 15, pagerPosition: 'bottom' }}
+      <SearchPanel
+        updateQuery={updateQuery}
+        updateSearch={updateSearch}
+        pageCount={pageCount}
+        filter={agentFilter}
+        showSearchBar={true}
+        disableSearch={disabledRows}
       />
+      <DarkenWhileDisabled className={disabledRows ? 'disabled' : ''}>
+        <Table
+          columns={[
+            {
+              key: 'hostname',
+              headerText: 'Hostname',
+              isSortable: true,
+            },
+            {
+              key: 'addr',
+              headerText: 'Address',
+              isSortable: true,
+              render: renderAddressCell,
+            },
+            {
+              key: 'labelsList',
+              headerText: 'Labels',
+              render: ({ labelsList }) => (
+                <ClickableLabelCell
+                  labels={labelsList}
+                  onClick={onAgentLabelClick}
+                />
+              ),
+            },
+            {
+              altKey: 'connect-btn',
+              render: server =>
+                renderConnectCell(
+                  () => getSshLogins(server.uri),
+                  login => connect(server, login)
+                ),
+            },
+          ]}
+          customSort={customSort}
+          emptyText={
+            fetchAttempt.status === 'success'
+              ? 'No servers found'
+              : 'Searching…'
```

Comment:
BTW, is sorting by address something that you plan to implement or should this column just not be sortable?

Reply:
The web ui doesn't support address sort so it seemed to have snuck in there. Disabling.